test/simp_result.lean

```diff
@@ -0,0 +1,138 @@
+import tactic.simp_result
+import data.equiv.basic
+
+open tactic
+
+-- Check that we can walk.
+example : true :=
+by { simp_result { trivial } }
+
+-- Comparison without `dsimp_result`:
+example : true :=
+begin
+  exact (id trivial),
+  (do `(id trivial) ← result, skip),
+  success_if_fail { (do `(trivial) ← result, skip) },
+end
+
+-- Check that `dsimp_result` removes unnecessary `id`s.
+example : true :=
+begin
+  dsimp_result { exact (id trivial) },
+  success_if_fail { (do `(id trivial) ← result, skip) },
+  (do `(trivial) ← result, skip),
+end
+
+-- Comparison without `dsimp_result`:
+example (a : ℕ) (b : list ℕ) (h : b.length < a) : ℕ :=
+begin
+  revert a,
+  intros a h,
+  exact 0,
+  (do `((λ (h : list.length _ < _), 0) _) ← result, skip),
+  success_if_fail { (do `(0) ← result, skip) },
+end
+
+-- Check that `dsimp_result` does beta-reductions after `revert`.
+example (a : ℕ) (b : list ℕ) (h : b.length < a) : ℕ :=
+begin
+  dsimp_result
+  { revert a,
+    intros a h,
+    exact 0, },
+  success_if_fail { (do `((λ (h : list.length _ < _), 0) _) ← result, skip), },
+  (do `(0) ← result, skip),
+end
+
+meta def guard_result_pp (s : string) : tactic unit :=
```

Comment:
I would really prefer if the tests didn't compare the string output.  When a test fails, I typically add `set_option pp.all true` for debugging and then the tests would fail even more.

```suggestion
meta def guard_result (p : parse texpr) : tactic unit :=
do r ← result, guard_expr_eq r p
```

Reply:
I just saw that this might get hairy when the result contains metavariables.  So my suggestion doesn't really work for all the cases here.

Reply:
I guess in `guard_result_pp` I could manually set the `pp` options, so the test won't be affected by user settings?

Reply:
I've made this slightly more robust, but not tried to completely defend against `pp` changes.